src/coreclr/vm/prestub.cpp

```diff
@@ -3772,7 +3824,7 @@ PCODE DynamicHelperFixup(TransitionBlock * pTransitionBlock, TADDR * pCell, DWOR
 
     _ASSERTE(pImportSection->EntrySize == sizeof(TADDR));
 
-    COUNT_T index = (rva - pImportSection->Section.VirtualAddress) / sizeof(TADDR);
+    COUNT_T index = (rva - pImportSection->Section.VirtualAddress) / pImportSection->EntrySize;
```

Comment:
The `_ASSERTE` above won't let us in here unless it's `sizeof(TADDR)` which suggests either the assert needs to go away or this divisor shouldn't change. I'm guessing the former, but then it's not clear to me whether this will always work if entrysize isn't what we expect it to be. (I'm not clear on how we get here yet)

Reply:
Also, `GetDataRva` appears to be able to return 0 in at least two scenarios, is this going to work correctly if `rva` is 0?

Reply:
This should be fine. I moved to use a divide instruction so that we handle out data format correctly even though we don't actually expect to see anything other than pointer sized entries here.